include/libdnf/rpm/nevra.hpp

```diff
@@ -149,6 +149,39 @@ inline void Nevra::set_arch(std::string && arch) {
 }
 
 
+/// Create a full nevra string (always contains epoch) from an object
+template <typename T>
+inline std::string to_full_nevra_string(const T & obj) {
+    auto epoch = obj.get_epoch();
+    if (epoch.empty()) {
+        epoch = "0";
+    }
+    return obj.get_name() + "-" + epoch + ":" + obj.get_version() + "-" + obj.get_release() + "." + obj.get_arch();
```

Comment:
I am curious how the construct will be optimized. I see 1 allocation, 8 reallotations, 17 copy operations.

Reply:
Replaced with reserve() and append().